sdks/java/core/src/main/java/org/apache/beam/sdk/io/Read.java

```diff
@@ -104,8 +105,7 @@ private Bounded(@Nullable String name, BoundedSource<T> source) {
       source.validate();
 
       return PCollection.<T>createPrimitiveOutputInternal(input.getPipeline(),
-          WindowingStrategy.globalDefault(), IsBounded.BOUNDED)
-          .setCoder(getDefaultOutputCoder());
+          WindowingStrategy.globalDefault(), IsBounded.BOUNDED);
```

Comment:
Could you also add a pair of unit tests to test this - where we successfully read (using a TestPipeline) from a bounded/unbounded source that does not override getDefaultOutputCoder, and the returned type of the source does not have a default coder, and instead the user (well, the test) calls .setCoder() on the resulting PCollection explicitly?